lib/openhab/dsl/types/datetime.rb

```diff
@@ -0,0 +1,339 @@
+
+        attr_reader :datetime
+
+        #
+        # Create a new DateTime instance wrapping an OpenHAB DateTimeType
+        #
+        # @param [Java::org::openhab::core::library::types::DateTimeType] datetime The DateTimeType instance to
+        #   delegate to, or an object that can be converted to a DateTimeType
+        #
+        def initialize(datetime)
+          @datetime = case datetime
+                      when DateTimeType
+                        datetime
+                      when ZonedDateTime
+                        DateTimeType.new(datetime)
+                      else
+                        datetime_type_from(datetime)
+                      end
+        end
+
+        #
+        # Compare thes DateTime object to another
+        #
+        # @param [Object] other Other object to compare against
+        #
+        # @return [Integer] -1, 0 or 1 depending on the outcome
+        #
+        def <=>(other)
+          logger.trace("Comparing self (#{self.class}) to #{other} (#{other.class})")
+          case other
+          when DateTime, DateTimeType, DateTimeItem
+            zoned_date_time.to_instant.compare_to(other.zoned_date_time.to_instant)
+          when TimeOfDay::TimeOfDay, TimeOfDay::TimeOfDayRangeElement
+            to_tod <=> other
+          else
+            self <=> DateTime.new(other)
+          end
+        end
+
+        #
+        # Adds another object to this DateTime
+        #
+        # @param [Object] other Object to add to this. Can be a Numeric, another DateTime/Time/DateTimeType, a
+        #   Duration or a String that can be parsed into a DateTimeType or Time object
+        #
+        # @return [DateTime] A new DateTime object representing the result of the calculation
+        #
+        # rubocop: disable Metrics/MethodLength
+        # Disabled line length - case dispatch pattern
+        def +(other)
+          logger.trace("Adding #{other} (#{other.class}) to #{self}")
+          case other
+          when Numeric
+            DateTime.new(to_time + other)
+          when DateTime, Time
+            self + other.to_f
+          when DateTimeType, String
+            self + DateTime.new(other).to_f
+          when Duration
+            DateTime.new(zoned_date_time.plus(other))
+          end
+        end
+
+        #
+        # Subtracts another object from this DateTime
+        #
+        # @param [Object] other Object to subtract fom this. Can be a Numeric, another DateTime/Time/DateTimeType, a
+        #   Duration or a String that can be parsed into a DateTimeType or Time object
+        #
+        # @return [DateTime, Float] A new DateTime object representing the result of the calculation, or a Float
+        #   representing the time difference in seconds if the subtraction is between two time objects
+        #
+        def -(other)
+          logger.trace("Subtracting #{other} (#{other.class}) from self")
+          case other
```

Comment:
same here.. one line check and probably need to break them out into methods in the cases.

Reply:
Can + and - share the same private methods with just the operator passed in?

Reply:
In ruby Time they work somewhat differently: + only accepts Numerics and always return a new Time object. - Does the same for Numerics, but also accepts another Time object and returns the time difference as a Float. I was trying to mimic that as much as possible, but with som extended functionality. That's why I had do write them as separate methods.

Reply:
ok, no need to overcomplicate it then..